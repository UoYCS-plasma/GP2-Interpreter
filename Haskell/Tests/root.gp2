Main = root ; swap

root () [
	(n1, empty # blue)
	(n2, empty # red)
	(n3, empty # grey)
|
	(e1, n1, n2, empty)
	(e2, n2, n3, empty)
] => [
	(n1 (R), empty # blue)
	(n2, empty # red)
	(n3, empty # grey)
|
	(e1, n1, n2, empty)
	(e2, n2, n3, empty)
]
interface = {n1, n2, n3}

swap () [
	(n1 (R), empty # blue)
	(n2, empty # red)
|
	(e1, n1, n2, empty)
] => [
	(n1, empty # blue)
	(n2 (R), empty # red)
|
	(e1, n1, n2, empty)
]
interface = {n1, n2}

