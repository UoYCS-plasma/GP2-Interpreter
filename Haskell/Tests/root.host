[
	(n1, empty # blue)
	(n2, empty # red)
	(n3, empty # grey)
|
	(e1, n1, n2, empty)
	(e2, n2, n3, empty)
]
