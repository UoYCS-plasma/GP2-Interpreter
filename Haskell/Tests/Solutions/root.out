[
 (n1, empty # grey)
 (n2 (R), empty # red)
 (n3, empty # blue)
|
 (e4, n2, n1, empty)
 (e5, n3, n2, empty)
]
